testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/TestRun.java

```diff
@@ -337,7 +338,9 @@ private Environment getEnvironment()
                 if (System.getenv("CONTINUOUS_INTEGRATION") != null) {
                     container.withEnv("CONTINUOUS_INTEGRATION", "true");
                 }
-                container
+
+                // Install Java distribution if necessary
+                jdkProvider.applyTo(container)
```

Comment:
Java packages aren't small, so this will take time (and may have reliability issues)
What prevents us from baking the JDKs on the images themselves, like we did previously? It seemed to have worked for us great.

Reply:
@findepi hence the configurable download path so it can be cached on the CI to minimize the impact. Base provided JDK stays. This PR makes it easier to run with some non-standard JDK without a need to rebuild _every_ docker image.

Reply:
I need this PR to test compatibility with JDK 21. I don't want to update images yet, and keeping multiple flavours doesn't make sense either. This won't replace existing "baked in JDK".

Reply:
@findepi as for the time, it takes around 10-15s to download JDK package (Temurin 20, around 200 MBs)